chaste_codegen/_math_functions.py

```diff
@@ -78,3 +78,20 @@ def fdiff(self, argindex=1):
 # MATH_FUNC_SYMPY_MAPPING provides a mapping from our specified math functions back to sympy versions.
 # This can be used to put sympy function into an expression or evaluation. e.g. `expr.subs(MATH_FUNC_SYMPY_MAPPING)`.
 MATH_FUNC_SYMPY_MAPPING = {abs_: Abs, acos_: acos, cos_: cos, exp_: exp, sin_: sin, sqrt_: sqrt}
+
+
+def subs_math_func_placeholders(expr):
+    """ Substitutes the placeholder math functions in expr for their corresponding Sympy functions
+    :param expr: sympy expression
+
+    Example:
+    >> str(expr)
+    '2.0 * exp_(V)'
+    >> subs_math_func_placeholders(expr)
+    '2.0 * exp(V)'
+
+    :return: expr with all placeholder functions replaced by sympy functions.
+    """
+    for placeholder_func, sympy_func in MATH_FUNC_SYMPY_MAPPING.items():
+        expr = expr.replace(placeholder_func, sympy_func)
```

Comment:
Is there a command that will replace multiple things at once? Would probably be faster to only traverse the expression tree a single time?

Reply:
There is it's called `subs` and it is **much much slower** mainly because it can also work on more complex expressions (both to find and replace). There is also `xreplace` but that doesn't work when the thing to swap is a function.